### PR TITLE
initial implementation of reading bona fide status from idp

### PR DESCRIPTION
**Pull request by opqdonut**
for #2142

This implementation has been tested on remsdev against the
elizir-czech test authentication service.

# Definition of Done / Review checklist

## Reviewability
- [x] link to issue
- [x] consider adding screenshots for ease of review

## Documentation
- [ ] update changelog if necessary

## Testing
- [ ] complex logic is unit tested
- [ ] valuable features are integration / browser / acceptance tested automatically

## Follow-up
- [ ] new tasks are created for pending or remaining tasks
- [x] no critical TODOs left to implement

### CODE REVIEW

**Comment by opqdonut** [comment]
Some questions for reviewers:
- Is `:bonafide` a good key in user attributes? Should we use something else?
- Do you think this needs more tests? I can always write tests that use mock JWTs but that won't guarantee compatibility with the actual tokens.
- There is still some refactoring & implementing to do, but I think this is a mergeable piece.

**Comment by opqdonut** [comment]
commit https://github.com/CSCfi/rems/pull/2398/commits/3f66dfb8b6d145369d034c0a1977bd598821e262 tested against elixir-czech login

File: src/clj/rems/ga4gh.clj [hunk 1]
```diff
@@ -34,10 +37,12 @@
    :sub s/Str
    :exp s/Int
    :iat s/Int
-   :scope (s/eq "openid") ;; could also list multiple space-separated scopes, as long as one is "openid"
+   (s/optional-key :scope) s/Str ;; Embedded Document Tokens "MUST NOT contain 'openid'"
```
Comment (Macroz) [inline src/clj/rems/ga4gh.clj:40]: This would be the right place for `;`

**Review by Macroz — APPROVED** [review]

File: src/clj/rems/ga4gh.clj (hunk 1 above)
Comment (opqdonut) [reply src/clj/rems/ga4gh.clj:40]: I don't like how emacs indents `;`....

Comment (opqdonut) [reply src/clj/rems/ga4gh.clj:40]: hmm no wait that's changed, it no longer indents the `;` to a fixed column

Comment (Macroz) [reply src/clj/rems/ga4gh.clj:40]: it's only if you put it on a newline?

**Pull request merged**